llm/ext_server/ext_server.cpp

```diff
@@ -148,11 +153,10 @@ void llama_server_stop() {
   assert(llama != NULL);
   // TODO - too verbose, remove once things are solid
   LOG_TEE("requesting llama server shutdown\n");
-  ext_server_running = false;
-
   // unblocks the update_slots() loop so it can clean up and exit
   llama->request_cancel(0);
-
+  llama->queue_results.remove_waiting_task_id(0);
```

Comment:
Would be worth checking if either of these are needed anymore

Reply:
I'll try removing it and see if there are any problems shutting down while requests are in flight.

Reply:
Good call.  This is no longer needed to get shutdown to happen cleaning.  Tested idle, as well as `^C` to the server while a prompt response was streaming and it quickly and gracefully shutdown in both cases.

Reply:
Actually, maybe I spoke too soon.  The signaling works on mac, but it gets stuck on linux if I try to kill while it's processing.  Investigating if it's related to removing this or some other signaling bug I need to figure out...

Reply:
Nope, there's still some other unrelated signaling problem on linux when trying to shutdown when a request is in flight.  I'll need to work on this some more before I can drop the draft status.